src/expr/src/scalar/func.rs

```diff
@@ -6601,15 +6676,15 @@ impl VariadicFunc {
             .nullable(false),
             SplitPart => ScalarType::String.nullable(in_nullable),
             RegexpMatch => ScalarType::Array(Box::new(ScalarType::String)).nullable(true),
-            HmacString | HmacBytes => ScalarType::Bytes.nullable(true),
-            ErrorIfNull => input_types[0].scalar_type.clone().nullable(false),
-            DateBinTimestamp => ScalarType::Timestamp.nullable(true),
-            DateBinTimestampTz => ScalarType::TimestampTz.nullable(true),
+            HmacString | HmacBytes => ScalarType::Bytes.nullable(in_nullable),
+            ErrorIfNull => input_types[0].scalar_type.clone().nullable(in_nullable),
```

Comment:
I'm not sure how `error_if_null` can ever produce a `null` output (even if the input is `null`). Can you give a counter-example?

Reply:
Nope---you're right!